LiskHQ/lisk-elements
# Fix not to expose custom external library definition - Closes #912

### What was the problem?
Custom external library definition that we have was used in the input of exposed function.
Since those custom definition doesn't get compiled, it will fail on compilation

### How did I fix it?
Move the production `types` into each folder, and it will be packaged to the npm repository.
Added `/// <reference path="..." />`, which is not recommended in the docs. 
https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

However, it seems like it's the only way to reference the custom external library definition

### How to test it?
`npm run build`
and check if the `validation.d.ts` and includes the reference tag

### Review checklist

* The PR resolves #912 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated
